content/articles/Servlet-Filter-API/index.md

```diff
@@ -0,0 +1,307 @@
+
+### Introduction.
+When an application is pre-and post-processed, the object being invoked is filtered. Filters are configured and used when filtering activities are performed. The filter interface is found in the `javax.servlet` package and to create a `Filter` class, we have to `import javax.servlet` and implement the `Filter` interface. several filters can be created to perform different tasks according to the user and developer specifications. 
+
+### Prerequisites
+- An open-source Java editor, such as Eclipse.Get it [here](https://www.eclipse.org/downloads/packages/release/2021-06/r).
+- a web browser installed.
+- Server. In my case, I will be using TomCat version 9. Download it from their official [website](https://tomcat.apache.org/)
+- basic knowledge of Java is needed.
+- Introduction to a servlet is needed.
+
+### Table of contents:
+
+- [What are Servlet Filters and How Do They Work?](#what-are-servlet-filters-and-how-do-they-work?)
+  - [servlet filter definition](#servlet-filter-definition)	
+  - [How Servlet Filters Works](#how-servlet-filters-works.)
+- [Filter usage and its advantages](#filter-usage-and-its-advantages)
+
+  - [Filter usage](#filter-usage)
+
+  - [Advantages](#advantages)	
+
+- [Filter interface](#filter-interface)	
+- [FilterChain interface](#filterchain-interface)
+- [FilterConfig interface](#filterconfig-interface)
+- [Authentication Example using filter](#authentication-example-using-filter)
+- [Conclusion](#conclusion)

```

Comment:
You forgot to mention the goals/key takeaways of this article. Have it before the table of contets.